### ISSUE TITLE
Unnecessary join: foreign key same as property name

### ISSUE THREAD

**Issue by emrosenf** [mikro-orm/mikro-orm#249]
**Describe the bug**

I am migrating a project where the foreign key in the database is not suffixed with `Id`. For example, the database table looks like this:

```sql
CREATE TABLE "Profile" (
  "id" TEXT NOT NULL   ,
  "user" TEXT    REFERENCES "User"(id) ON DELETE SET NULL,
  PRIMARY KEY ("id")
)
```

The problem is that when you query this class like so:

```typescript
em.find(Profile, { user: { $in: [userId] }, })
```

The code below in `ObjectCriteriaNode` will trigger. It will see that `user` is a non-scalar field and join in the `User` table, even though I'm really just trying to query on an indexed foreign key that happens to also be called `user` in the database.

```typescript
export class ObjectCriteriaNode extends CriteriaNode {

  static create(metadata: MetadataStorage, entityName: string, payload: Dictionary, parent?: CriteriaNode, key?: string): ObjectCriteriaNode {
    const node = new ObjectCriteriaNode(metadata, entityName, parent, key);
    const meta = metadata.get(entityName, false, false);
    node.payload = Object.keys(payload).reduce((o, item) => {
      // prop is defined because "user" is a defined relationship
      // as well as the name of the database field that holds the foreign key
      const prop = meta && meta.properties[item];
      const childEntity = prop && prop.reference !== ReferenceType.SCALAR ? prop.type : entityName;
      o[item] = CriteriaNode.create(metadata, childEntity, payload[item], node, item);

      return o;
    }, {});

    return node;
  }
```

**Comment by B4nan** [comment]
You need to provide this information either via [custom naming strategy](https://mikro-orm.io/docs/naming-strategy/), or manually in each property:

```typescript
@Entity()
class Profile {
  @OneToOne({ joinColumn: 'user' })
  user?: User;
}
```

**Comment by B4nan** [comment]
Btw that part of `CriteriaNode` is correct, if there is a problem, it will be in the processing part, not when creating. But first, you need to fix your entity definitions - if your db do not follow current naming strategy, nothing can work correctly. 

If you have the database schema consistent, I'd suggest to go for custom naming strategy. Btw there is `EntityCaseNamingStrategy` that follows this convention on join columns, and afaik it is working good this way, so I don't think there is actually a bug here, it will be most probably just misconfiguration. Here is an example project that uses this strategy: https://github.com/lookfirst/mikro-example

**Comment by B4nan** [comment]
I think you should be fine with the `EntityCaseNamingStrategy`, just noted you also have the table named `Profile`, not `profile`, which is exactly how that strategy behaves. 

What `EntityCaseNamingStrategy` does:
- table name is same as entity name (`Profile` -> `Profile`)
- column name is same as property name (`email` -> `email`)
- join column is same as property name (`user` -> `user`)
- for M:N pivot tables it will use entity names (`Ent1/Ent2` -> `Ent1_to_Ent2`)

To use it, simply provide it when initializing the ORM:

```typescript
const orm = await MikroORM.init({
  // ...
  namingStrategy: EntityCaseNamingStrategy,
});
```

**Comment by emrosenf** [comment]
I am using the `EntityCaseNamingStrategy` and my db schema is consistent with it.

I am preferring to use `ManyToOne` over `OneToOne` because I want to minimize the unwanted join.

Here is the `Profile` class:

```typescript
@Entity()
class Profile {
  @ManyToOne({ entity: () => User, fieldName: 'user', cascade: [Cascade.MERGE], nullable: true, })
  user?: User;
}
```

Here is my query call:

```typescript
em.find(Profile, { user: { $in: [userId] }, })
```

And here is the query output:
```bash
select `e0`.* from `Profile` as `e0` left join `User` as `e1` on `e0`.`user` = `e1`.`id` where `e0`.`user` in (...)
```

I do not believe the join to `User` is necessary, and I notice that the query builder changes to add the `User` table in the code snippet I posted. It appears that in the reduce function, it loops through the properties and it sees that the queried field, `user,` is not a `SCALAR` and so it pulls in the `User` table.

**Comment by B4nan** [comment]
Yeah it is non needed indeed. Will try to look into this next week, the auto joining feature is pretty new and there might be similar issues here and there...

**Comment by emrosenf** [comment]
Thank you, I would really appreciate that. You are correct that it seems to be `autoJoin` related. Let me know if I can help. When I'm done with all of these questions, we should be able to add a few more examples to the docs.

**Comment by B4nan** [comment]
Any contribution to the docs is more than welcome :)

**Comment by B4nan** [comment]
Fixed in `3.0.0-rc.1`.

**Comment by emrosenf** [comment]
Just wanted to thank you (for the other issues too), I have everything working the way I want now!
